--- /usr/share/redmine/app/views/projects/settings/_repositories.html.erb	2012-05-13 22:10:22.000000000 +0300
+++ _repositories.html.erb	2012-06-08 13:26:04.000000000 +0300
@@ -1,4 +1,7 @@
 <% if @project.repositories.any? %>
+<% unless ScmConfig.configured? %>
+  <div class="warning"><%= l(:text_scm_creator_not_configured_warning) %></div>
+<% end %>
 <table class="list">
   <thead>
     <tr>
@@ -25,7 +28,7 @@
         <%= link_to(l(:button_delete), repository_path(repository),
             :confirm => l(:text_are_you_sure),
             :method  => :delete,
-            :class   => 'icon icon-del') %>
+            :class   => 'icon icon-del') unless repository.created_with_scm && ScmConfig['deny_delete'] %>
       <% end %>
       </td>
     </tr>
