--- /usr/share/redmine/app/views/projects/settings/_repositories.html.erb	2016-10-10 10:48:13.000000000 +0300
+++ app/views/projects/settings/_repositories.html.erb	2016-12-22 14:14:01.226879759 +0200
@@ -1,3 +1,6 @@
+<% unless ScmConfig.configured? %>
+  <div class="warning"><%= l(:text_scm_creator_not_configured_warning) %></div>
+<% end %>
 <% if @project.repositories.any? %>
 <table class="list">
   <thead>
@@ -18,14 +21,22 @@
       </td>
       <td><%= checked_image repository.is_default? %></td>
       <td><%= repository.scm_name %></td>
+      <% interface = SCMCreator.interface(repository) %>
+      <% if repository.created_with_scm && interface && interface.enabled? %>
+      <td><%= h(interface.external_url(repository) || repository.url) %></td>
+      <% else %>
       <td><%= repository.url %></td>
+      <% end %>
       <td class="buttons">
       <% if User.current.allowed_to?(:manage_repository, @project) %>
         <%= link_to(l(:label_user_plural), committers_repository_path(repository),
              :class => 'icon icon-user') %>
         <%= link_to(l(:button_edit), edit_repository_path(repository),
              :class => 'icon icon-edit') %>
-        <%= delete_link repository_path(repository) %>
+        <%= link_to(l(:button_delete), repository_path(repository),
+            :confirm => l(:text_are_you_sure),
+            :method  => :delete,
+            :class   => 'icon icon-del') unless repository.created_with_scm && ScmConfig['deny_delete'] %>
       <% end %>
       </td>
     </tr>
